notebooks/upload_new_dataset.py

```
@@ -183,40 +231,39 @@ def convert_to_standard_format(df: pl.DataFrame) -> pl.DataFrame:
-        return df
-
-    # Get unique sectors
-    df = df.with_columns(pl.col('Sector').map_elements(to_snake_case, return_dtype=pl.String))
-    unique_sectors = df.select('Sector').unique().to_series(0).to_list()
-    unique_sectors = [s for s in unique_sectors if s is not None]
-
-    # If only one sector, just rename Value column to that sector
-    if len(unique_sectors) == 1:
-        sector = unique_sectors[0]
+def pivot_by_compound_id(df: pl.DataFrame, metric_col: str) -> pl.DataFrame:
+    """Convert dataframe to have compound identifiers as columns."""
+    # Get unique compound IDs
+    unique_ids = df.select('CompoundID').unique().to_series(0).to_list()
+    unique_ids = [s for s in unique_ids if s is not None]
+    # If only one compound ID, just rename Value column to that ID
+    if len(unique_ids) == 1:
+        compound_id = unique_ids[0]
         return df.with_columns(
-            pl.col('Value').alias(sector)
-        ).drop(['Sector', 'Value'])
+            pl.col('Value').alias(compound_id)
+        ).drop([metric_col, 'Quantity', 'Value', 'CompoundID'])
 
-    # Get dimension columns (all except Sector and Value)
-    dim_cols = [col for col in df.columns if col not in ['Sector', 'Value']]
+    # Get dimension columns (all except those used for pivoting and the value)
+    dim_cols = [col for col in df.columns if col not in [metric_col, 'Quantity', 'Value', 'CompoundID']]
 
-    # Ensure no null sectors
+    # Ensure no null compound IDs
     df = df.with_columns(
-        pl.when(pl.col('Sector').is_null())
+        pl.when(pl.col('CompoundID').is_null())
           .then(pl.lit("unknown"))
-          .otherwise(pl.col('Sector'))
-          .alias('Sector')
+          .otherwise(pl.col('CompoundID'))
+          .alias('CompoundID')
     )
 
-    # Pivot to have sectors as columns
+
+    # Pivot to have compound IDs as columns
+    try:
+        df = df.with_columns(pl.col('Value').cast(pl.Float64))
+    finally:
```

The intention is to ignore exceptions? Maybe better to ignore a specific one, or handle it, maybe logging it?

---

Exactly. Good point